### ISSUE TITLE
Implement the base Clean Architecture with MVVM: presentation/app layer

### ISSUE THREAD

**Issue by luongvo** [nimblehq/kmm-templates#14]
## Why

The template should provide the initial base architecture to start. The selected architecture for the shared, Android and iOS modules is [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) with MVVM.

## Who Benefits?

Developers

**Comment by phongvhd93** [comment]
This needs to be evaluated. Most of KMM-IC are now sharing the Domain and Data layer only. From my exp sharing ViewModel is quite tricky

**Comment by kaungkhantsoe** [comment]
I can implement the viewModel inside the `android` module and only share the `Data` and `Domain`. What is your thought on this @luongvo ?

**Comment by luongvo** [comment]
@kaungkhantsoe Yes, let's go with a native VM in each Android and iOS module now. cc @phongvhd93

**Comment by luongvo** [comment]
Hi @kaungkhantsoe. I'm finishing the task of reusing the Android app module from Android Templates #29. Please postpone this task until I finish my one to re-check and revise the plan for further necessary updates for the Android module. 🙏

**Comment by kaungkhantsoe** [comment]
Got it @luongvo 🙏
